analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ParameterTypeShouldMatchRouteTypeConstraint.razor

```diff
@@ -0,0 +1,73 @@
+@page "/route/{BoolParam:bool}"
+@page "/route/{BoolParam:int}"
+@page "/route/{BoolParam:int}/{DatetimeParam:datetime}"
+@page "/route/{DatetimeParam:datetime}"
+@page "/route/{DecimalParam:decimal}"
+@page "/route/{DoubleParam:double}"
+@page "/route/{FloatParam:float}"
+@page "/route/{GuidParam:guid}"
+@page "/route/{IntParam:int}"
+@page "/route/{LongParam:long}"
+@page "/route/{StringParam:string}"
+@page "/route/{OptionalStringParam:string?}"
```

Comment:
I would also try with an optional struct, such as `int` or `long`.
Do they work with nullable value types? 
That is: if you declare an `int?` on the property, is that compatible with both an `int` and an `int?` on the route constraint side?

Another question that would How do they combine with non-nullable reference types?
It would be worth to add such tests.

Reply:
> Do they work with nullable value types?

Yes, see [Query strings](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-7.0#query-strings). Although it is about query string it is still relevant for route param as well, especially this part:
> Component parameters supplied from the query string support the following types:
>
> - bool, DateTime, decimal, double, float, Guid, int, long, string.
> - Nullable variants of the preceding types.
> - Arrays of the preceding types, whether they're nullable or not nullable.

----

> That is: if you declare an int? on the property, is that compatible with both an int and an int? on the route constraint side?

The `?` on the route constraint side only specifies that the route parameter is optional. See in this section, [Route parameters](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-7.0#route-parameters). It is in the end not a constraint.
In this case, when mapping to non-nullable value, the property will be its default (e.g: 0 for int), for a nullable the value will be `null`.

----

> Another question that would How do they combine with non-nullable reference types?

As stated in the previous answer, it will use the default of the type.